pkg/event/target/kafka.go

```diff
@@ -32,6 +32,21 @@ type KafkaArgs struct {
 	Enable  bool        `json:"enable"`
 	Brokers []xnet.Host `json:"brokers"`
 	Topic   string      `json:"topic"`
+	Net struct {
+		SASL struct {
+			User      string `json:"username"`
+			Password  string `json:"password"`
+			Handshake bool   `json:"handshake"`
+			Enable    bool   `json:"enable"`
+		} `json:"sasl"`
+		TLS struct {
+			Enable bool  `json:"enable"`
+			Config struct {
+				InsecureSkipVerify bool `json:"insecureskipverify"`
+				ClientAuth         int  `json:"clientauth"`
+			} `json:"tlsconfig"`
+		} `json:"tls"`
+	} `json:"net"`
```

Comment:
1. Nesting of SASL/TLS configuration is not good idea.  It will lead to complicated configuration file.
2. Nested TLS.Config is a bad idea.
3. Is Kafka used with SASL and TLS independently?

Reply:
Keep simplified configuration change like below

```golang
// KafkaArgs - Kafka target arguments.
type KafkaArgs struct {
	Enable  bool        `json:"enable"`
	Brokers []xnet.Host `json:"brokers"`
	Topic   string      `json:"topic"`
	TLS     struct {
		Enable     bool               `json:"enable"`
		SkipVerify bool               `json:"skipVerify"`
		ClientAuth tls.ClientAuthType `json:"clientAuth"`
	} `json:"tls"`
	SASL struct {
		Enable    bool   `json:"enable"`
		Username  string `json:"username"`
		Password  string `json:"password"`
		Handshake bool   `json:"handshake"`
	} `json:"sasl"`
}
```

Reply:
+1 @balamurugana 

However, are we planning to switch the client library to segmentio/kafka-go? The library does not have authentication yet.

Reply:
If it's not then we don't need to migrate, we can migrate when it implements authentication. @annanay25 But they implement TLS based encryption https://github.com/segmentio/kafka-go/issues/109#issuecomment-403086031